pkg/kp/pcstore/consul_store.go

```diff
@@ -193,6 +198,136 @@ func (s *consulStore) FindWhereLabeled(podID types.PodID,
 	return ret, nil
 }
 
+// WatchPodCluster watches the podClusterTree for changes.
+// It will return a blocking channel on which the client can read
+// WatchedPodCluster objects. The goroutine maintaining the watch will block on
+// writing to this channel so it's up to the caller to read it with haste.
+func (s *consulStore) Watch(quit <-chan struct{}) <-chan WatchedPodCluster {
+	inCh := make(chan api.KVPairs)
+	outCh := make(chan WatchedPodCluster)
+	errChan := make(chan error, 1)
+
+	go consulutil.WatchPrefix(podClusterTree, s, inCh, quit, errChan)
+
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-errChan:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
+			}
+		}
+	}()
+
+	go func() {
+		var kvp api.KVPairs
+		for {
+			select {
+			case kvp = <-inCh:
+				if kvp == nil {
+					// nothing to do
+					continue
+				}
+			case <-quit:
+				return
+			}
+
+			pcs, err := kvpsToPC(kvp)
+			if err != nil {
+				select {
+				case <-quit:
+					return
+				case outCh <- WatchedPodCluster{PodCluster: nil, Err: err}:
+				}
+			}
+
+			for _, pc := range pcs {
+				select {
+				case outCh <- WatchedPodCluster{PodCluster: &pc, Err: nil}:
+				case <-quit:
+					return
+				}
+			}
+		}
+	}()
+
+	return outCh
+}
+
+// WatchPodCluster implements a watch for the Pod cluster at _id_
+// It will return a blocking channel on which the client can read
+// WatchedPodCluster objects. The goroutine maintaining the watch will block on
+// writing to this channel so it's up to the caller to read it with haste.
+func (s *consulStore) WatchPodCluster(id fields.ID, quit <-chan struct{}) <-chan WatchedPodCluster {
+	inCh := make(chan *api.KVPair)
+	outCh := make(chan WatchedPodCluster)
+	errChan := make(chan error, 1)
+	key := path.Join(podClusterTree, string(id))
+
+	go func() {
+		for {
+			select {
+			case <-quit:
+				return
+			case err := <-errChan:
+				s.logger.WithError(err).Errorf("WatchPrefix returned error, recovered.")
```

Comment:
damn github is displaying our comments in the wrong place... we're both commenting on line 274 but our comment is being associated with  218 because they're the same line!!!! whoaaaaaaaaaaaaa